File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Netherlands.m
//
//  UIImage+Netherlands.m
//  CountriesFlags
//  

#import "Countries/UIImage+Netherlands.h"

@implementation UIImage (Netherlands)

+(UIImage * _Nullable) flagNetherlandsWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:30.0f/255.0f
                                         green:71.0f/255.0f
                                          blue:133.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:173.0f/255.0f
                                        green:29.0f/255.0f
                                         blue:37.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Netherlands.swift
//
//  UIImage+Netherlands.swift
//  CountriesFlags
//

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_the_Netherlands "wikipedia"
     
     Generates the flag of Netherlands.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Netherlands.
     */
    internal static func flagNetherlands(with size: CGSize) -> UIImage? {
        let colorWhite = UIColor.white;
        let colorBlue = UIColor(red: 30.0/255.0,
                                green: 71.0/255.0,
                                blue: 133.0/255.0,
                                alpha: 1.0)
        let colorRed = UIColor(red: 173.0/255.0,
                               green: 29.0/255.0,
                               blue: 37.0/255.0,
                               alpha: 1.0)
        
        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
        
        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))
        
        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))
        
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return image
    }
}
